test/units/parsing/test_splitter.py

```diff
@@ -173,3 +173,12 @@ def test_split_args_error(args):
 def test_parse_kv_error(args):
     with pytest.raises(AnsibleParserError):
         parse_kv(args)
+
+
+@pytest.mark.parametrize("args", (
+    'creates=something thisisextra',
+    'something thisisextra',
+))
+def test_parse_kv_remainder_error(args: str) -> None:
+    with pytest.raises(ValueError, match='Invalid k=v detected'):
```

Comment:
Let's make it obvious that it's not a simple substring match.
```suggestion
    with pytest.raises(ValueError, match=r'^Invalid k=v detected: .*$'):
```